### PR TITLE
[Transformation] Add BroadcastConstRangeReplacement and enable offline SmartReshape

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
- BroadcastConstRangeReplacement transformation needed to enable model reshapeability based on:
https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/front/broadcast_with_range.py
It replaces Constant range of numbers by Range op as input data to Broadcast op.

Other requested changes - discussed offline with @GlebKazantaev:
- Make the BroadcastConstRangeReplacement a part of SmartReshape transformations
- Enable and apply SmartReshape transformations as a part of offline transformations (apply_moc_transformations)

### Tickets:
 - 76721

### CODE REVIEW

**Comment by GlebKazantaev** [comment]
In general looks good.
Waiting for reshape test results.

**Review by jane-intel — APPROVED** [review]
Good job!
I have only one minor comment about unnecessary Squeeze

File: src/common/transformations/src/transformations/smart_reshape/broadcast_const_range_replacement.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            return false;
+
+        const auto data_elem_type = data_const_out.get_element_type();
+        const auto target_dim_index = std::distance(const_node_shape.cbegin(), std::find(const_node_shape.cbegin(), const_node_shape.cend(), elem_count));
+        const int64_t target_dim_neg_index = target_dim_index - const_node_shape.size();
+
+        const auto axis_node = ngraph::opset8::Constant::create(ngraph::element::i32, {}, {0});
+        const auto target_dim_index_node = ngraph::opset8::Constant::create(ngraph::element::i64, {}, {target_dim_neg_index});
+        const auto gather_dim = std::make_shared<ngraph::opset8::Gather>(target_shape_out, target_dim_index_node, axis_node);
+        const auto scalar_gather_dim = std::make_shared<ngraph::opset8::Squeeze>(gather_dim);
```
[the remark this answers is not in the document]
Comment (mitruska) [reply src/common/transformations/src/transformations/smart_reshape/broadcast_const_range_replacement.cpp]: You're right, the Squeeze was redundant. Removed, thanks!

**Comment by mitruska** [comment]
BroadcastConstRangeReplacement transformation will be added in the separate PR https://github.com/openvinotoolkit/openvino/pull/10138
without offline SmartReshape enablement.

**Pull request closed without merging**